packages/terminal/src/browser/terminal-frontend-contribution.ts

```diff
@@ -189,11 +189,39 @@ export class TerminalFrontendContribution implements TerminalService, CommandCon
         const widget = this.shell.currentWidget;
         if (widget instanceof TerminalWidget) {
             this.setCurrentTerminal(widget);
+            this.setLastUsedTerminal(widget);
         } else if (!this._currentTerminal || !this._currentTerminal.isVisible) {
             this.setCurrentTerminal(undefined);
         }
     }
 
+    // IDs of the most recently used terminals
+    protected mostRecentlyUsedTerminalIds: string[] = [];
+
+    protected getLastUsedTerminalId(): string | undefined {
+        return this.mostRecentlyUsedTerminalIds[this.mostRecentlyUsedTerminalIds.length - 1];
+    }
+
+    get lastUsedTerminal(): TerminalWidget | undefined {
+        const id = this.getLastUsedTerminalId();
+        if (id) {
+            return this.getById(id);
+        }
+    }
+
+    protected setLastUsedTerminal(lastUsed: TerminalWidget): void {
+        const lastUsedTerminalId = lastUsed.id;
+        if (!this.lastUsedTerminal || this.getLastUsedTerminalId() !== lastUsedTerminalId) {
+            this.mostRecentlyUsedTerminalIds.push(lastUsedTerminalId);
+            lastUsed.onTerminalDidClose(() => {
```

Comment:
maybe `onDispose`? i.e. if a terminal process was not ever opened we still should clean up `mostRecentlyUsedTerminalIds` then a widget is disposed

Reply:
done.